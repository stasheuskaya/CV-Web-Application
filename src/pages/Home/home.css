.startScreen {
    background: linear-gradient(to top, #41414180, rgba(65, 65, 65, .5)), url(../../assets/images/backgroundStart.png) no-repeat center/contain;
    color: var(--main-text-color-white);
    align-items: center;
    padding-right: 20px;
    padding-left: 20px;
    height: 100vh;
    width: auto;
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center center;
    margin-bottom: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
