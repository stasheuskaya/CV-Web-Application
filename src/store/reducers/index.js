import { combineReducers } from "redux";
import { skillsReducer } from "./skills_reducer";
import { educationsReducer } from "./educations_reducer";

const allReducers = combineReducers({
    skills: skillsReducer,
    educations: educationsReducer
})

export default allReducers;