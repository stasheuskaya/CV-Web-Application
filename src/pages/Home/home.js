import './home.css';
import Button from "../../components/Button/button";
import PhotoBox from "../../components/PhotoBox/photo_box";
import userAvatar from '../../assets/images/userAvatar.png';
import { useNavigate } from "react-router-dom";

export default function Home() {
    let navigate = useNavigate(); 
    const routeChange = (path) => () => {
        navigate(path);
    }

    return (
        <>
            <div className="startScreen">
                <div>
                <PhotoBox 
                    avatar={<img src={userAvatar} className="userAvatar" alt="userAvatar"/>}
                    name="Yuliya Stasheuskaya"
                    userSummary="Programmer. Creative. Innovator"
                    description={"Web developer with 6 months of experience in designing websites. \n Recognized for leadership, collaborative abilities when working in team settings."}
                />
                <Button 
                    className='startButton' 
                    title='Know more'
                    onClickHandler={routeChange("/inner")}
                />
                </div>
            </div>
        </>
    )
}
