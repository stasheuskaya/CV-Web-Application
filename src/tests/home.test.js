import renderer from "react-test-renderer";
import Home from "../pages/Home/home"
import { Provider } from 'react-redux';
import { store } from "../store/store"
import { BrowserRouter, Routes, Route } from "react-router-dom";

describe('Home Page', () => {
	beforeAll(() => {
        window.history.pushState({}, '', `/`);
	});

    it('renders content', () => {
		expect(renderer.create(
			<Provider store={store}>
				<BrowserRouter>
                    <Routes>
                        <Route path="/" element={<Home />}/>
                    </Routes>
                </BrowserRouter>
			</Provider>)).toMatchSnapshot();
	});
});
