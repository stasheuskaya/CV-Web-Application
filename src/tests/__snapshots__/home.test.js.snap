// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders content 1`] = `
<div
  className="startScreen"
>
  <div>
    <div
      className="photoBox"
    >
      <div
        className="userImage"
      >
        <img
          alt="userAvatar"
          className="userAvatar"
          src="userAvatar.png"
        />
      </div>
      <h1
        className="userName"
        name="Yuliya Stasheuskaya"
      >
        Yuliya Stasheuskaya
      </h1>
      <h2
        className="userSummary"
      >
        Programmer. Creative. Innovator
      </h2>
      <div
        className="userDescription"
        style={
          Object {
            "whiteSpace": "pre-line",
          }
        }
      >
        Web developer with 6 months of experience in designing websites. 
 Recognized for leadership, collaborative abilities when working in team settings.
      </div>
    </div>
    <button
      className="startButton"
      onClick={[Function]}
    >
      <div
        className="icon"
      />
      Know more
    </button>
  </div>
</div>
`;
